workflow/executor/executor.go

```diff
@@ -531,24 +531,44 @@ func (we *WorkflowExecutor) SaveParameters(ctx context.Context) error {
 }
 
 // SaveLogs saves logs
-func (we *WorkflowExecutor) SaveLogs(ctx context.Context) (*wfv1.Artifact, error) {
+func (we *WorkflowExecutor) SaveLogs(ctx context.Context) ([]wfv1.Artifact, []error) {
```

Comment:
I don't think this should return mulitple errors - it not canonical Go - instead - could you add `containerName` as an argument and just call many times?

Reply:
i moved the iteration over the `MainContainer` "outside" to the `waitContainer` func. it makes the function less concise.
another option i just tried out (haven't pushed) - 
```go
func (we *WorkflowExecutor) SaveAllLogs(ctx context.Context) ([]wfv1.Artifact, error) {
	logArts := make([]wfv1.Artifact, 0)
	if we.Template.SaveLogsAsArtifact() {
		return logArts, nil
	}

	// Saving logs
	log.Infof("Saving logs")
	tempLogsDir := "/tmp/argo/outputs/logs"
	err := os.MkdirAll(tempLogsDir, os.ModePerm)
	if err != nil {
		return logArts, err
	}

	containerNames := we.Template.GetMainContainerNames()
	for _, containerName := range containerNames {
		art, err := we.SaveLogs(ctx, tempLogsDir, containerName)
		if err != nil {
			we.AddError(err)
		} else if art != nil {
			logArts = append(logArts, *art)
		}
	}

	return logArts, nil
}
```

this will keep the iteration and error handling inside the WorkflowExecutor code, which is a bit different from how it is handled in the other calls from `waitContainer`. do you think this is preferable?